src/domino.py

```diff
@@ -0,0 +1,174 @@
+        @param module_threshold: the threshold for considering a putative module as final module (optional)
+        @param singularity: if True, run using the Singularity container instead of the Docker container (optional)
+        """
+        # Assuming defaults are: use_cache=true
+
+        if not network or not active_genes or not output_file:
+            raise ValueError('Required DOMINO arguments are missing')
+
+        work_dir = '/spras'
+
+        # Each volume is a tuple (source, destination)
+        volumes = list()
+
+        bind_path, network_file = prepare_volume(network, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, node_file = prepare_volume(active_genes, work_dir)
+        volumes.append(bind_path)
+
+        # Use its --output_folder argument to set the output file prefix to specify an absolute path and prefix
+        out_dir = Path(output_file).parent
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+
+        ########
+
+        bind_path, mapped_slices_dir = prepare_volume('slices_dir', work_dir)
+        volumes.append(bind_path)
+        slices_file = mapped_slices_dir + '/slices.txt'
+
+        slicer_command = ['slicer',
+            '--network_file', network_file,
+            '--output_file', slices_file]
+
+        container_framework = 'singularity' if singularity else 'docker'
+        slicer_out = run_container(container_framework,
+                            'otjohnson/domino',
+                            slicer_command,
+                            volumes,
+                            work_dir)
+        print(slicer_out)
+
+
+        ########
+
+
+        # Make the Python command to run within the container
+        command = ['domino',
+                   '--active_genes_files', node_file,
+                   '--network_file', network_file,
+                   '--slices_file', slices_file,
+                   '--output_folder', mapped_out_dir,
+                   '--parallelization', '1',
+                   '--visualization', 'true']
+
+        # Add optional arguments
+        if use_cache is not True:
+            command.extend(['-c', 'false'])
+        if slices_threshold is not None:
+            command.extend(['-sth', str(slices_threshold)])
+        if module_threshold is not None:
+            command.extend(['-mth', str(module_threshold)])
+
+        print('Running DOMINO with arguments: {}'.format(' '.join(command)), flush=True)
+
+        # container_framework = 'singularity' if singularity else 'docker'
+        out = run_container(container_framework,
+                            'otjohnson/domino',
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+
+
+        ########
+
+        slices_file.unlink(missing_ok=True)
```

Comment:
```suggestion
        Path(slices_file).unlink(missing_ok=True)
```
My Python editor is complaining that `slices_file` is a string, not a Path, so it doesn't have an `unlink` function. We can convert it to a Path here or earlier to use `unlink` on it.